### ISSUE TITLE
Can't connect to Kafka 2.3.0 brokers or load legacy-frontend

### ISSUE THREAD

**Issue by Dreeseaw** [singularity-data/risingwave#2968]
**Describe the bug**
Hello, I've been attempting to create a Kafka materialized source from a Kafka 2.3.0 broker. 

Query (some whitewashing, but I can ping the brokers and connect with the python connector). 
~~~
CREATE MATERIALIZED SOURCE IF NOT EXISTS kafka_table (
   h varchar,
   u varchar,
   p varchar,
   cet varchar,
   cec varchar,
   V varchar,
   sv varchar,
)
WITH (
   'connector'='kafka',
   'kafka.topic'='<topic>',
   'kafka.brokers'='<brokers addresses>',
   'kafka.scan.startup.mode'='latest',
   'kafka.time.offset'='140000000',
   'kafka.consumer.group'='<group_id>'
)
ROW FORMAT JSON;
~~~

Results in 
~~~
2022-06-02T16:01:42.978148Z DEBUG librdkafka: librdkafka: PROTOERR [thrd:<broker>:9092/bootstrap]: <broker>:9092/bootstrap: Protocol parse failure for ApiVersion v3(flex) at 3/6 (rd_kafka_handle_ApiVersion:2131) (incorrect broker.version.fallback?)
2022-06-02T16:01:42.978296Z DEBUG librdkafka: librdkafka: PROTOERR [thrd:<broker>:9092/bootstrap]: <broker>:9092/bootstrap: ApiArrayCnt -1 out of range

(repeated for each broker in the list)
~~~

Upon reading the docs, I found the tip saying Kafka connectors are broken at the moment. When attempting to run it, I get -
~~~
./risedev configure enable legacy-frontend

error: "legacy-frontend" isn't a valid value for '<COMPONENT>'
        [possible values: minio, prometheus-and-grafana, etcd, kafka, tracing, rust-components, dashboard, release, all-in-one, sanitizer]
~~~

When attempting to just run the given Docker image (MacOS with x86-64 Intel chip) -
~~~
➜  risingwave git:(main) docker run -it --pull=always -p 4566:4566 -p 5691:5691 ghcr.io/singularity-data/risingwave:latest playground

latest: Pulling from singularity-data/risingwave
Digest: sha256:1ae788d124a895f6e9477ffa5ccae7bbbb8ece2c396c2c228a70d9a7dd1bf150
Status: Image is up to date for ghcr.io/singularity-data/risingwave:latest
thread 'main' panicked at 'OS can't spawn worker thread: Operation not permitted (os error 1)', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/blocking/pool.rs:264:29
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
~~~

**To Reproduce**
Broker's Kafka Version: 2.3.0
Broker's librdkafka-dev Version: 1.1.0~1confluent5.3.4-1
Use Query above
Mac Version: BigSur 11.6, 2.3 GHz 8-Core Intel Core i9
Docker version: Docker Desktop 4.1.1 (69879)

**Expected behavior**
A materialized stream to be created and queryable.

**Additional context**
Recently installed Rust to run the project locally, if that helps at all.

**Comment by neverchanje** [comment]
Legacy frontend is not intended to be run so  you can see it as it doesn't exist. Pls run the lastest version of RisingWave because we don't test the correctness of the previous versions.

**Comment by Dreeseaw** [comment]
@neverchanje I did a git pull on the repo just now - (top commit)
~~~
commit a3e618ed5fda77abdd2bfadfb2bd171b605f8567 (HEAD -> main, origin/main, origin/HEAD)
Author: nanderstabel <nanderstabel@gmail.com>
Date:   Thu Jun 2 16:31:34 2022 +0200
~~~

I am receiving the exact same error as before. Also, I would like to validate that [sources](https://www.risingwave.dev/docs/latest/sources/) is out of date documentation, as like you've said above, you shouldn't be using the legacy frontend.

Thanks for the response.

**Comment by skyzh** [comment]
The docs is outdated 😂

You may follow https://www.risingwave.dev/docs/latest/perform-real-time-ad-performance-analysis/ instead.

**Comment by neverchanje** [comment]
Thank you for reporting the doc problem, we will soon fix it!

**Comment by hengm3467** [comment]
@Dreeseaw The note about legacy frontend in Sources was just removed. Thanks for letting us know.

**Comment by Dreeseaw** [comment]
@neverchanje @hengm3467 Gotcha, then this starts to make A LOT more sense. 

@skyzh Thanks for the permissions issue link, it worked great! The full working command I ended up with was
~~~
sudo docker run -it --pull=always -p 4566:4566 -p 5691:5691 --privileged --security-opt="seccomp=unconfined" ghcr.io/singularity-data/risingwave:latest playground
~~~

Closing the issue as the system I'm trying to integrate with runs Kafka 2.3.0, and I don't have the ability to upgrade it (and supporting Kafka 3.x & 2.x seems a little much to ask). Thank you all for the quick assistance!
